### PR TITLE
 Docs: API overview text clarity & kv2 mention... #16746

### PR DESCRIPTION
**Pull request by aphorise**
Corrected text and terminology. Relates to #6378 & should allow for closure of that issue too.
Screenshot 2022-08-16 at 19 23 20

### CODE REVIEW

File: website/content/api-docs/index.mdx [hunk 1]
```diff
@@ -123,9 +126,9 @@ endpoints that consume parameters use `POST` instead and put the parameters in
 the request body.
 
 You can list secrets as well. To do this, either issue a GET with the query
-parameter `list=true`, or you can use the `LIST` HTTP verb. For the `kv`
-backend, listing is allowed on directories only, and returns the keys in the
-given directory:
+parameter `list=true`, or you use the `LIST` HTTP verb. For the `kv` secrets
```
Comment (benashz) [inline website/content/api-docs/index.mdx:129]: You may want to put caveat portion of this statement in italics.

File: website/content/api-docs/index.mdx [hunk 2]
```diff
@@ -192,12 +195,15 @@ addition, the Vault SDK always adds this header to every request.
 
 ## Help
 
-To retrieve the help for any API within Vault, including mounted backends, auth
+To retrieve the help for any API within Vault, including mounted engine, auth
```
Comment (benashz) [inline website/content/api-docs/index.mdx:198]:
```suggestion
To retrieve the help for any API within Vault, including mounted engines, auth
```

File: website/content/api-docs/index.mdx [hunk 3]
```diff
@@ -192,12 +195,15 @@ addition, the Vault SDK always adds this header to every request.
...
 methods, etc. then append `?help=1` to any URL. If you have valid permission to
-access the path, then the help text will be return a markdown-formatted block in the `help` attribute of the response.
+access the path, then the help text will be return a markdown-formatted block
```
Comment (benashz) [inline website/content/api-docs/index.mdx:200]:
```suggestion
access the path, then the help text will be returned as a markdown-formatted block 
```

File: website/content/api-docs/index.mdx (hunk 1 above)
Comment (aphorise) [reply website/content/api-docs/index.mdx:129]: @benashz do you mean the `or` or which portion do you think should be made italic?

File: website/content/api-docs/index.mdx [hunk 4]
```diff
@@ -90,14 +86,21 @@ $ curl \
...
+The demonstration below are using the `KVv1` secrets engine, which is a 
+simple Key/Value store. Please read the API documentation of secret engine or 
+authentication engine for detailed information on their use. 
 
-Different plugins implement different APIs according to their functionality.
-The examples below are created with the `KVv1` backend, which acts like a very
-simple Key/Value store. Read the documentation for a particular backend for
-detailed information on its API; this simply provides a general overview.
+There are documented difference in the path schemes as well as other 
+functionality which may be applicable as for example with [`KVv2` compared to `KVv1`].
```
[the remark this answers is not in the document]
Comment (aphorise) [reply website/content/api-docs/index.mdx:104]: There was some feedback originally on how difference in KV1 / KV2 is not explicit spelt out. The original does seem a bit generic - but please do suggest any alterative that you may see as more suitable.

File: website/content/api-docs/index.mdx [hunk 5]
```diff
@@ -90,14 +86,21 @@ $ curl \
...
+appropriately with its response and does not require it from the clients 
+request.
+
+Authentication and Secret engines, that are also refer to as backends or 
+mounts, differ in their implementation of APIs according to functionality and 
+how they're authored.
+
+The demonstration below are using the `KVv1` secrets engine, which is a 
+simple Key/Value store. Please read the API documentation of secret engine or 
+authentication engine for detailed information on their use.
```
[the remark this answers is not in the document]
Comment (aphorise) [reply website/content/api-docs/index.mdx:100]: I will correct this on the next commit.

File: website/content/api-docs/index.mdx [hunk 6]
```diff
@@ -90,14 +86,21 @@ $ curl \
 
 ## API Operations
 
-With few documented exceptions, all request body data and response data from
-Vault is via JSON. Vault will set the `Content-Type` header appropriately but
-does not require that clients set it.
+All request data, body and response data to and from Vault is JSON except in a 
+few exceptional cases that are documented. Vault sets the `Content-Type` header
```
Comment (benashz) [inline website/content/api-docs/index.mdx:90]: I would omit the last bit about being `documented` unless  you can provide a link to said documentation.

File: website/content/api-docs/index.mdx [hunk 7]
```diff
@@ -90,14 +86,21 @@ $ curl \
...
-Vault is via JSON. Vault will set the `Content-Type` header appropriately but
-does not require that clients set it.
+All request data, body and response data to and from Vault is JSON except in a 
+few exceptional cases that are documented. Vault sets the `Content-Type` header 
+appropriately with its response and does not require it from the clients 
+request.
+
+Authentication and Secret engines, that are also referred to as backends or 
+mounts, differ in their implementation of APIs according to functionality and 
+how they're authored.
```
Comment (benashz) [inline website/content/api-docs/index.mdx:96]: This sentence is hard to parse. I am not sure what it is  trying to be  conveyed by this statement. Are you trying to say that each API may have a unique implementation/design?

File: website/content/api-docs/index.mdx (hunk 4 above)
[the remark this answers is not in the document]
Comment (benashz) [reply website/content/api-docs/index.mdx:104]: You could provide a table of differences, or comparable usage examples for each.

File: website/content/api-docs/index.mdx [hunk 8]
```diff
@@ -21,12 +21,6 @@ is, `sys/` routes) change very infrequently, but various secrets engines/auth
 methods/etc. sometimes have minor changes to accommodate new features as
 they're developed.
```
Comment (benashz) [inline website/content/api-docs/index.mdx:23]: The request PATH difference seems to be one of the most obvious differences between v1 and v2, but this change seem to be dropping this bit of information all together?

File: website/content/api-docs/index.mdx (hunk 1 above)
Comment (benashz) [reply website/content/api-docs/index.mdx:129]: I think we can ignore this comment. I may have misinterpreted it.

File: website/content/api-docs/index.mdx (hunk 6 above)
Comment (aphorise) [reply website/content/api-docs/index.mdx:90]: Ok let me see what I can do here - the original and current docs do note the same exception but you are right without linking it's not helpful - all those other cases are by user request in any case (I was thinking like some PKI endpoints that provide binary).

File: website/content/api-docs/index.mdx [hunk 9]
```diff
@@ -90,14 +86,21 @@ $ curl \
...
+few exceptional cases that are documented. Vault sets the `Content-Type` header 
+appropriately with its response and does not require it from the clients 
+request.
+
+Authentication and Secret engines, that are also refer to as backends or 
+mounts, differ in their implementation of APIs according to functionality and 
+how they're authored.
+
+The demonstration below are using the `KVv1` secrets engine, which is a 
+simple Key/Value store. Please read the API documentation of secret engine or
```
[the remark this answers is not in the document]
Comment (aphorise) [reply website/content/api-docs/index.mdx]: added KVv1 API links

File: website/content/api-docs/index.mdx (hunk 8 above)
Comment (aphorise) [reply website/content/api-docs/index.mdx:23]: @benashz I've added a specific note on URI path differences and features.

File: website/content/api-docs/index.mdx (hunk 4 above)
[the remark this answers is not in the document]
Comment (aphorise) [reply website/content/api-docs/index.mdx:96]: @benashz I think as an intro the way it is at present can suffice and just listing differences KVv2 in any table could be the bulk of the page (especially since KV2 has so much more) rather than this page just being a quick overview and intro into how API calls are made.

**Comment by aphorise** [comment]
Hey @benashz can you kindly recheck again.

**Comment by aphorise** [comment]
Hey @benashz further to your feedback I've added the KV2 example explicitly.

I've also adjusted the grammer a bit as there are:
 - `query` should be `query string` 
 - over use of the word `via` in various places to mean `in` or `using` or `with`, etc.
 - over use of the word `to` - especially in some places eg:
     >Vault tries **to** (1) adhere **to** (2) these whenever possible, but in some cases may not -- feel free **to** (3) 
file a bug in that case **to** (4) point our attention **to** (5) it!

@yhyakuna I'd appreciate your eyes on this too (and anyone else).

**Review by yhyakuna — APPROVED** [review]
Thank you for catching those.
